serialization/include/archives/SimdJsonInputArchive.h

```diff
@@ -65,6 +65,7 @@ class SimdJsonInputArchive
   SimdJsonInputArchive& Serialize(T& output)
   {
     static_assert(!sizeof(T), "can only parse to std::string");
+    return *this; // gcc wants it
```

Comment:
I compiled with gcc without this line

Reply:
It gives warning

Reply:
Hm actually
```
/home/roma/my/skymp/skymp-arch/serialization/include/archives/SimdJsonInputArchive.h: In member function ‘SimdJsonInputArchive& SimdJsonInputArchive::Serialize(T&)’:
/home/roma/my/skymp/skymp-arch/serialization/include/archives/SimdJsonInputArchive.h:69:3: error: no return statement in function returning non-void [-Werror=return-type]
   69 |   }
      |   ^
cc1plus: some warnings being treated as errors
```
It's a warning that's treated as error

Reply:
```
ubuntu@9b94c15f0dde:/home/roma/my/skymp/skymp-arch$ c++ --version
c++ (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```